cmd/tests/tracing_module_test.go

```diff
@@ -128,6 +99,232 @@ func TestTracingClient_DoesNotInterfereWithHTTPModule(t *testing.T) {
+}
+
+func TestTracingInstrumentHTTP_Jaeger(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	var gotRequests int64
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+		assert.NotEmpty(t, r.Header.Get("uber-trace-id"))
+		assert.Len(t, r.Header.Get("uber-trace-id"), 45)
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "jaeger",
+		})
+
+		export default function () {
+			http.del("HTTPBIN_IP_URL/tracing");
+			http.get("HTTPBIN_IP_URL/tracing");
+			http.head("HTTPBIN_IP_URL/tracing");
+			http.options("HTTPBIN_IP_URL/tracing");
+			http.patch("HTTPBIN_IP_URL/tracing");
+			http.post("HTTPBIN_IP_URL/tracing");
+			http.put("HTTPBIN_IP_URL/tracing");
+			http.request("GET", "HTTPBIN_IP_URL/tracing");
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Equal(t, int64(8), atomic.LoadInt64(&gotRequests))
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	assertHasTraceIDMetadata(t, jsonResults)
+}
+
+func TestTracingInstrumentHTTP_FillsParams(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	var gotRequests int64
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
+
+		assert.NotEmpty(t, r.Header.Get("X-Test-Header"))
+		assert.Equal(t, "test", r.Header.Get("X-Test-Header"))
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "w3c",
+		})
+
+		const testHeaders = {
+			"X-Test-Header": "test",
+		}
+
+		export default function () {
+			http.del("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.get("HTTPBIN_IP_URL/tracing", { headers: testHeaders });
+			http.head("HTTPBIN_IP_URL/tracing", { headers: testHeaders });
+			http.options("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.patch("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.post("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.put("HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+			http.request("GET", "HTTPBIN_IP_URL/tracing", null, { headers: testHeaders });
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Equal(t, int64(8), atomic.LoadInt64(&gotRequests))
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	assertHasTraceIDMetadata(t, jsonResults)
```

Comment:
Kind of the same comment about this not checking that all of those have tracing ids.

Maybe `assertHasTraceIDMetadata` can have a number argument that is how many it expects :shrug:. 

Or to take the expected url(s) (as vararg not required argument)  and check that all of those have trace id metadata sample.